internal/provider/data_source_gitlab_project_membership.go

```diff
@@ -0,0 +1,188 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+var _ = registerDataSource("gitlab_project_membership", func() *schema.Resource {
+	return &schema.Resource{
+		Description: "Provide details about a list of project members in the gitlab provider. The results include id, username, name and more about the requested members.\n\n" +
+			"> **Note**: exactly one of project_id or full_path must be provided.",
+
+		ReadContext: dataSourceGitlabProjectMembershipRead,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Description: "The ID of the project.",
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Optional:    true,
+				ConflictsWith: []string{
```

Comment:
`ExactlyOneOf` would be more correct here, since it enforces that one must be set. See: https://github.com/gitlabhq/terraform-provider-gitlab/blob/b1521811a1e99391801e69b19287565a4918257d/internal/provider/resource_gitlab_deploy_token.go#L33

Reply:
Should I also update the group membership with this as well?